app/controllers/mailgun_webhooks_controller.rb

```diff
@@ -92,12 +92,15 @@ def create_incoming_email
   end
 
   def update_outgoing_email_status
-    email_to_update = OutgoingEmail.find_by(message_id: params.dig("event-data", "message", "headers", "message-id"))
+    message_id = params.dig("event-data", "message", "headers", "message-id")
+    email_to_update = OutgoingEmail.find_by(message_id: message_id)
```

Comment:
In the super cool future, we could encode _which_ model we expect to match on, before we communicate to Mailgun. That way this can be 100% deterministic. Completely not a big deal, just a dream.

Reply:
Alternatively, one day, we could make a table called `MailgunStatus` and store the message ID in it, and then only update that one. Then we'd need to JOIN onto it when we are showing an OutgoingEmail's status. Useful? Too useless to be worth the bother? I leave that question unanswered.